packages/core/test/unit/application.test.ts

```diff
@@ -8,10 +8,60 @@ import {Application, Server, Component} from '../../index';
 import {Context, Constructor} from '@loopback/context';
 
 describe('Application', () => {
+  describe('controller binding', () => {
+    let app: Application;
+    class MyController {}
+
+    beforeEach(givenApp);
+
+    it('binds a controller', () => {
+      const binding = app.controller(MyController);
+      expect(binding.tags.has('controller')).to.be.true();
+      expect(binding.key).to.equal('controllers.MyController');
+      expect(app.findByTag('controller')[0]).to.equal(binding);
+    });
+
+    it('binds a controller with custom name', () => {
+      const binding = app.controller(MyController, 'my-controller');
+      expect(binding.tags.has('controller')).to.be.true();
+      expect(binding.key).to.equal('controllers.my-controller');
+      expect(app.findByTag('controller')[0]).to.equal(binding);
+    });
+
+    function givenApp() {
+      app = new Application();
+    }
+  });
+
+  describe('component binding', () => {
+    let app: Application;
+    class MyController {}
+    class MyComponent implements Component {
+      controllers = [MyController];
+    }
+
+    beforeEach(givenApp);
+
+    it('binds a component', () => {
+      app.component(MyComponent);
+      expect(app.findByTag('component').length).to.equal(1);
+    });
+
+    it('binds a component', () => {
+      app.component(MyComponent, 'my-component');
+      expect(app.find('components.my-component').length).to.equal(1);
```

Comment:
Similarly here, asserting on array length is almost never a good thing to do. If you don't want to compare bindings, then at least use "have.length" assertion.

An example of a helpful assertion failure:

```
> expect([1,2,3]).to.have.length(2)
AssertionError: expected Array [ 1, 2, 3 ] to have property length of 2 (got 3)
```

An example of an unhelpful error:

```
> expect([1,2,3].length).to.have.equal(2)
AssertionError: expected 3 to be 2
```